### PR TITLE
feat(aom): add new resource alarm action rule

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/aom' TESTARGS='-run TestAccAlarmActionRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aom -v -run TestAccAlarmActionRule_basic -timeout 360m -parallel 4
=== RUN   TestAccAlarmActionRule_basic
=== PAUSE TestAccAlarmActionRule_basic
=== CONT  TestAccAlarmActionRule_basic
--- PASS: TestAccAlarmActionRule_basic (30.80s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aom       30.901s
```

### CODE REVIEW

File: huaweicloud/services/aom/resource_huaweicloud_aom_alarm_action_rule.go [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+		if len(rawArray) == 0 {
+			return nil
+		}
+
+		rst := make([]map[string]interface{}, len(rawArray))
+		for i, v := range rawArray {
+			raw := v.(map[string]interface{})
+			rst[i] = map[string]interface{}{
+				"topic_urn": utils.ValueIngoreEmpty(raw["topic_urn"]),
+				"name":      utils.ValueIngoreEmpty(raw["name"]),
```
Comment (houpeng80) [inline huaweicloud/services/aom/resource_huaweicloud_aom_alarm_action_rule.go]: this line is uesless

Comment (Jason-Zhang9309) [reply huaweicloud/services/aom/resource_huaweicloud_aom_alarm_action_rule.go]: fixed

**Comment by ShiChangkuo** [comment]
@Jason-Zhang9309 please add a preCheck for the acceptance as the resource is only supported in cn-east-3.

**Comment by Jason-Zhang9309** [comment]
> @Jason-Zhang9309 please add a preCheck for the acceptance as the resource is only supported in cn-east-3.

sure

**Comment by Jason-Zhang9309** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/aom' TESTARGS='-run TestAccAlarmActionRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/aom -v -run TestAccAlarmActionRule_basic -timeout 360m -parallel 4
=== RUN   TestAccAlarmActionRule_basic
=== PAUSE TestAccAlarmActionRule_basic
=== CONT  TestAccAlarmActionRule_basic
--- PASS: TestAccAlarmActionRule_basic (34.20s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/aom       34.376s
```

**Pull request merged**